src/proto/script_host.proto

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright 2020 Nick Mertin
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package bowler.script_host;
+
+import "git_f_s.proto";
+
+option java_multiple_files = true;
+option java_package = "com.commonwealthrobotics.proto.script_host";
+
+// Next: 5
+// The client message type
+message SessionClientMessage {
+    // The content of the message
+    oneof message {
+        // Request to run a script
+        RunRequest run_request = 1;
+
+        // Response to a confirmation request
+        ConfirmationResponse confirmation_response = 2;
+
+        // Response to a credentials request
+        CredentialsResponse credentials_response = 3;
+
+        // Error in execution of a request
+        RequestError error = 4;
+    }
+}
+
+// Next: 5
+// Client message to run a script
+message RunRequest {
+    // The unique ID of the request
+    int64 request_id = 1;
+
+    // The file to run
+    bowler.git_f_s.FileSpec file = 2;
+
+    // Dev'd dependencies
+    repeated bowler.git_f_s.ProjectSpec devs = 3;
+
+    // Script environment variables
+    map<string, string> environment = 4;
+}
+
+// Next: 3
+// Client message to respond to a confirmation request
+message ConfirmationResponse {
+    // The ID of the request
+    int64 request_id = 1;
+
+    // The value of the response
+    ConfirmationValue response = 2;
+}
+
+// Next: 2
+// Value of a confirmation response
+enum ConfirmationValue {
+    // User responded 'Allow'
+    ALLOWED = 0;
+
+    // User responded 'Deny'
+    DENIED = 1;
+}
+
+// Next: 4
+// Client message to fulfill a credentials request
+message CredentialsResponse {
+    // The ID of the request which is being fulfilled
+    int64 request_id = 1;
+
+    // The credentials to use
+    oneof credentials {
+        // Username and password pair
+        BasicCredentials basic = 2;
+
+        // OAuth
+        OAuthCredentials oauth = 3;
+    }
+}
+
+// Next: 4
+// Credentials implementation for a username and password pair
+message BasicCredentials {
+    // The username
+    string username = 1;
+
+    // The password
+    string password = 2;
+
+    // Optional 2FA response
+    string two_factor = 3;
```

Comment:
I think the way 2FA works is that after you sign in you are prompted for a 2FA code, so I think we need to properly model
1. Sign in with username & password/PAT
2. Turns out we need a 2FA code
3. Ask the user for 2FA code and get that response (i.e. I think this needs to be another message)
4. Give GitHub the 2FA code

I guess it's one bidirectional stream, though, so we could just send this `BasicCredentials` message again? I'm not sure how we respond to the first one, though (error isn't right; we need a "give me 2FA" response).

Reply:
Implemented as a separate request/response instead.